metrics-core/src/lib.rs

```diff
@@ -74,7 +257,7 @@ pub trait Recorder {
     /// counters and gauges usually have slightly different modes of operation.
     ///
     /// For the sake of flexibility on the exporter side, both are provided.
-    fn record_counter<K: Into<Key>>(&mut self, key: K, value: u64);
+    fn record_counter(&mut self, key: Key, value: u64);
```

Comment:
I imagine that you removed the generic here so that `Recorder` can potentially be used as a trait object?

Reply:
I originally did it that way because of the facade crate, yeah, wanting to be able to do `&'static Recorder`.

I haven't switched it over yet because I'm not sure it makes sense to force implementations to take multiple values at a time for histograms (`record_histogram(&mut self, key: Key, values: &[u64])`) since that was a design choice to help _exporters_ and code actually _recording_ metrics typically is sending a single value at a time.